department-of-veterans-affairs/caseflow
# Reassign to Judge

Resolves APPEALS-21598

Description
Backend for Reassign to Judge is created. When we click on Assign button in the Reassign to Judge modal, the Caseflow and VACOLS databases will be updated. When we reassign a case from one VLJ to another VLJ, the task table from the Caseflow database and BRIEFF table from the VACOLS database will be updated and the PRIORLOC table from the VACOLS database will have a new row inserted. 

 Acceptance Criteria
- [x] When Assign button is clicked, caseflow and vacols DB will be updated to complete the selected task action of “Re-assign to Judge”
priorloc and brief VACOLS tables will be updated with re-assignment 
Add a new row in the priorloc table when a location change is made on a Legacy Appeal 

Testing Plan
1. Switch the user as a Special case movement user : BVARDUNKLE
2. In search cases give the Veteran ID (Give the VBMS_ID from the legacy_appeals table as Veteran id in Caseflow for Legacy appeals)
<img width="836" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110078646/b0ea9f85-3a2b-4e17-a168-8980f41078cb">
3. Click on the case id and go to Case details page.
<img width="833" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110078646/e572a470-fc81-44c0-ae98-15de82d603db">
4. In the Currently active task action dropdown, select Case movement: cancel blocking task and advance to judge
<img width="836" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110078646/761613eb-81d7-467e-af4a-726ad54b8033">
5. In the reassign view give the reason and instructions and click on continue button
<img width="830" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110078646/615fc354-121e-4f9a-b3fd-21edc3742583">
6. In the Confirm cancellation modal select the Judge and give instructions and click on Cancel task and Reassign
<img width="848" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110078646/244c95cf-9bba-4f0a-85af-24f55ecbcb5c">
7. Now switch user as Rachael Judge and click on Queue
8.  In the Rachael Judge view click on Switch view and select Assign BVARERDMAN team case
9.  Now we can move this case to another judge, from the actions dropdown select Reassign to Judge
<img width="831" alt="image" src="https://github.com/department-of-veterans-affairs/caseflow/assets/110078646/3ccc74d8-23ea-4825-8d1c-2a2a099abd36">
10. If we click on Assign button from the Assign task modal, now the database tables from Caseflow and VACOLS will update


 



# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---

## Storybook Story
*For Frontend (Presentationa) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
